File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/callback/ParametersAndReturnsTestBoolean.jl
println("getBoolean")
if isdefined(Main, :getBoolean) && !isnothing(getBoolean)
	error("getBoolean already defined!")
end
getBoolean = callback("getBoolean")
println(typeof(getBoolean))
println(getBoolean)
if typeof(getBoolean) != Bool
	error("getBoolean not Bool!")
end
callback("setBoolean",getBoolean)

println("getBooleanVector")
if isdefined(Main, :getBooleanVector) && !isnothing(getBooleanVector)
	error("getBooleanVector already defined!")
end
getBooleanVector = callback("getBooleanVector")
println(eltype(getBooleanVector))
println(getBooleanVector)
if eltype(getBooleanVector) != Bool 
	error("getBooleanVector not Bool!")
end
callback("setBooleanVector",getBooleanVector)

println("getBooleanVectorAsList")
if isdefined(Main, :getBooleanVectorAsList) && !isnothing(getBooleanVectorAsList)
	error("getBooleanVectorAsList already defined!")
end
getBooleanVectorAsList = callback("getBooleanVectorAsList")
println(eltype(getBooleanVectorAsList))
println(getBooleanVectorAsList)
if eltype(getBooleanVectorAsList) != Bool
	error("getBooleanVectorAsList not Bool!")
end
callback("setBooleanVectorAsList",getBooleanVectorAsList)

println("getBooleanMatrix")
if isdefined(Main, :getBooleanMatrix) && !isnothing(getBooleanMatrix)
	error("getBooleanMatrix already defined!")
end
getBooleanMatrix = callback("getBooleanMatrix")
println(eltype(getBooleanMatrix))
println(getBooleanMatrix)
if eltype(getBooleanMatrix) != Bool
	error("getBooleanMatrix not Bool!")
end
callback("setBooleanMatrix",getBooleanMatrix)

println("getBooleanMatrixAsList")
if isdefined(Main, :getBooleanMatrixAsList) && !isnothing(getBooleanMatrixAsList)
	error("getBooleanMatrixAsList already defined!")
end
getBooleanMatrixAsList = callback("getBooleanMatrixAsList")
println(eltype(getBooleanMatrixAsList))
println(getBooleanMatrixAsList)
if eltype(getBooleanMatrixAsList) != Bool
	error("getBooleanMatrixAsList not Bool!")
end
callback("setBooleanMatrixAsList",getBooleanMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestBoolean.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getBoolean")
if(getBoolean ~= nil) then
	error("getBoolean already defined!")
end
getBoolean = callback("getBoolean")
getBooleanType = type(getBoolean)
print(getBooleanType)
print(getBoolean)
if(getBooleanType ~= "boolean") then
	error("getBoolean not Boolean!")
end
callback("setBoolean", getBoolean)

print("getBooleanVector")
if(getBooleanVector ~= nil) then
	error("getBooleanVector already defined!")
end
getBooleanVector = callback("getBooleanVector")
getBooleanVectorType = type(getBooleanVector[1])
print(getBooleanVectorType)
print(arraysToString(getBooleanVector))
if(getBooleanVectorType ~= "boolean") then
	error("getBooleanVector not Boolean!")
end
callback("setBooleanVector", getBooleanVector)

print("getBooleanVectorAsList")
if(getBooleanVectorAsList ~= nil) then
	error("getBooleanVectorAsList already defined!")
end
getBooleanVectorAsList = callback("getBooleanVectorAsList")
getBooleanVectorAsListType = type(getBooleanVectorAsList[1])
print(getBooleanVectorAsListType)
print(arraysToString(getBooleanVectorAsList))
if(getBooleanVectorAsListType ~= "boolean") then
	error("getBooleanVectorAsList not Boolean!")
end
callback("setBooleanVectorAsList", getBooleanVectorAsList)

print("getBooleanMatrix")
if(getBooleanMatrix ~= nil) then
	error("getBooleanMatrix already defined!")
end
getBooleanMatrix = callback("getBooleanMatrix")
getBooleanMatrixType = type(getBooleanMatrix[1][1])
print(getBooleanMatrixType)
print(arraysToString(getBooleanMatrix))
if(getBooleanMatrixType ~= "boolean") then
	error("getBooleanMatrix not Boolean!")
end
callback("setBooleanMatrix", getBooleanMatrix)

print("getBooleanMatrixAsList")
if(getBooleanMatrixAsList ~= nil) then
	error("getBooleanMatrixAsList already defined!")
end
getBooleanMatrixAsList = callback("getBooleanMatrixAsList")
getBooleanMatrixAsListType = type(getBooleanMatrixAsList[1][1])
print(getBooleanMatrixAsListType)
print(arraysToString(getBooleanMatrixAsList))
if(getBooleanMatrixAsListType ~= "boolean") then
	error("getBooleanMatrixAsList not Boolean!")
end
callback("setBooleanMatrixAsList", getBooleanMatrixAsList)
